packages/block-library/src/list/split-on-line-break.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { last } from 'lodash';
+
+/**
+ * Split the content of a paragraph on line breaks ('<br>') into sets of
+ * content, each representing a list item.
+ *
+ * The term "content" refers to a rich-text description in the form of a mixed
+ * array of strings and element-like objects (e.g. `{ type: strong, props: {
+ * children: […] } }`).
+ *
+ * @param {WPBlockChildren} fragments Rich-text content
+ * @return {Array<WPBlockChildren>} Array of rich-text content
+ */
+export default function splitOnLineBreak( fragments ) {
+	return fragments.reduce( ( acc, fragment, i, arr ) => {
+		// Skip if fragment is a line break
+		if ( fragment && fragment.type === 'br' ) {
+			return acc;
+		}
+
+		// If we've just skipped a line break, append the
+		// next fragment as a new item.
+		const prevFragment = i > 0 && arr[ i - 1 ];
+		if ( prevFragment && prevFragment.type === 'br' ) {
```

Comment:
Observation: We're piercing the abstraction of the `node` / `children` type here.

Still an open question as to whether that's okay / expected. But it does further solidify us into committing to this object shape.

Reply:
@aduth, before I add any comments to it, does 43ad834 look like a better compromise?

Reply:
Cool. Rectified in 8af5cc3.